src/System Application/App/Cryptography Management/src/CryptographyManagementImpl.Codeunit.al

```diff
@@ -270,6 +283,12 @@ codeunit 1279 "Cryptography Management Impl."
         HashAlgorithm.Dispose();
     end;
 
+    [NonDebuggable]
+    procedure GenerateHash(InputString: SecretText; "Key": SecretText; HashAlgorithmType: Option HMACMD5,HMACSHA1,HMACSHA256,HMACSHA384,HMACSHA512): Text
+    begin
+        exit(GenerateHash(InputString.Unwrap(), Key, HashAlgorithmType));
+    end;
+
     [NonDebuggable]
     procedure GenerateHash(InputString: Text; "Key": SecretText; HashAlgorithmType: Option HMACMD5,HMACSHA1,HMACSHA256,HMACSHA384,HMACSHA512): Text
```

Comment:
Update this function to take in SecretText for the InputString and unwrap the value only here. While the rest of the functions can pass either Text or SecretText until it gets to this.
Add overloads as needed.

We don't want to unwrap any SecretText outside of the location it is actually being used.

Reply:
Move the logic to the procedure with secret text and have the other procedure call it, so you dont have to duplicate the code.

Reply:
Remember to do the same for other procedures.